fastestimator/network.py

```diff
@@ -13,23 +13,27 @@
 # limitations under the License.
 # ==============================================================================
 from collections import ChainMap
-from typing import Any, Dict, Iterable, List, MutableMapping, Set, Union
+from typing import Any, Callable, Dict, Iterable, List, MutableMapping, Set, Union
 
 import tensorflow as tf
-
 import torch
+
 from fastestimator.op import TensorOp, get_current_ops, get_inputs_by_op, write_outputs_by_op
 from fastestimator.op.tensorop.model import ModelOp, UpdateOp
 from fastestimator.schedule import EpochScheduler, RepeatScheduler, Scheduler
 from fastestimator.util.util import NonContext, lcms, to_list
 
 
 class BaseNetwork:
-    def __init__(self, ops: Iterable[Union[TensorOp, Scheduler[TensorOp]]]):
+    def __init__(self,
+                 ops: Iterable[Union[TensorOp, Scheduler[TensorOp]]],
+                 models: Union[tf.keras.Model, torch.nn.Module]):
         self.ops = to_list(ops)
+        self.models = to_list(models)
```

Comment:
is the only reason that we have this variable to make it more convenient than looking through self.ops all the time?

Reply:
is so we should convert it to a cached functional attribute

Reply:
self.models is used to get the optimizer information (like signature epochs of all optimizers)

Reply:
but the information it encodes is redundant with the information encoded in self.ops right?

Reply:
done